python/GafferUI/SpreadsheetUI/_ProxyModels.py

```diff
@@ -64,16 +64,16 @@ def setSourceModel( self, model ) :
 
 		assert( isinstance( model, _PlugTableModel ) )
 
-		# _PlugTableModel only emits modelReset and dataChanged signals (for now)
+		# _PlugTableModel only emits modelReset and [header]DataChanged signals (for now)
 		# so we can avoid the headache of remapping the plethora of row/column signals.
 
 		oldModel = self.sourceModel()
 		if oldModel :
-			oldModel.dataChanged.disconnect( self.__dataChanged )
-			oldModel.modelReset.disconnect( self.modelReset )
+			oldModel.disconnect( self )
```

Comment:
I had to go look this one up again to be sure it was the right way round! I find it really disconcerting that Qt has overloads where the receiver is in the arg list _and_ overloads where the sender is in the arg list - I don't think I'm ever going to remember. As with my earlier comments for the selection proxy model, I don't think we actually need `setSourceModel()` anyway, so would be good to remove all this tracking code at some point (not in this PR though).

Reply:
Hehe, I know what you mean. I guess I just think of it as being the same as if it was a slot. 

Re: `setSourceModel`. I implemented it that was as its part of the [standard implementation](https://doc.qt.io/qt-5/qabstractproxymodel.html#setSourceModel) of a qt proxy model. Changing the pattern felt like it might come back to haunt us in the future. The selection proxy model doesn't exist in Qt, so we were at least fee to change that (though I originally had put it in to match the model proxy pattern for consistency).

Reply:
> Re: setSourceModel. I implemented it that way as its part of the standard implementation of a qt proxy model. Changing the pattern felt like it might come back to haunt us in the future.

Ah, OK, makes sense.